x/programs/runtime/utils.go

```diff
@@ -0,0 +1,18 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package runtime
+
+const None = byte(0)
+
+func Ok(data []byte) []byte {
```

Comment:
Use the type system instead of []byte.  Doing it this way makes it unclear when bytes are actually bytes or if they are an option or a result.

Reply:
I will probably just base my branch again on your `AddResult` instead of duplicating efforts